### ISSUE TITLE
Transaction details: refunding an inquiry should use the refund modal rather than redirect to the order screen

### ISSUE THREAD

**Issue by Jinksi** [Automattic/woocommerce-payments#7963]
### Description

Transaction refunds can now be handled directly on the transaction details screen, rather than via the order screen (see PR #7742). We should utilise this new refund modal to improve the UX for merchants wishing to resolve an inquiry by issuing a refund. Some copy changes will also be required (`You will be taken to the order` is no longer relevant).

This screen recording shows the current inquiry modal, with a button directing them to the order screen. It then demonstrates the new refund modal that should be used for this refund flow.

https://github.com/Automattic/woocommerce-payments/assets/3147296/d627bb8a-b336-4ddd-a4dd-01f5830c12f5

### Acceptance criteria

* When clicking "issue refund", the call-to-action button should show the refund modal and not redirect to the order screen

### Testing instructions

1. Create an inquiry on a transaction by using the card `4000000000001976` at checkout
2. View the transaction details page as a merchant
3. Click the **Issue refund** button

**Comment by haszari** [comment]
Added to backlog with `medium` priority. Inquiries are less common, but it would be simpler and better to use consistent modal flow for them too. 

Hopefully quick to implement. @Jinksi please add `good first issue` label if this is relatively straightforward.

**Comment by jessy-p** [comment]
@rogermattic  Please review the copy and modal changes.

### recording - suggested flow
https://github.com/user-attachments/assets/4af96cb7-8580-4d21-863e-3595237db917

### screenshot - changed informational modal
![Image](https://github.com/user-attachments/assets/8367215c-88cc-480d-ae65-5c16a07d3366)

Clicking on the Full Refund uses existing refund modal. Clicking on Partial refund takes you to the order.

**Comment by rogermattic** [comment]
Thank you for the ping @jessy-p ! Great job :)

I can confirm that the changes you applied are good to go. I only wondered if:
> Clicking on Partial refund takes you to the order.

1. Could we not only redirect the user to the order page but also scroll them directly to the Product section where the refund process happens, with the Refund options already expanded?

![Image](https://github.com/user-attachments/assets/67069967-10bd-4847-b43e-b619864f47b1)

2. Should we simplify and:

- remove the introductory text "Before proceeding, please take note of the following:" 
- remove the first icon
![Image](https://github.com/user-attachments/assets/55038148-9d27-4e06-affb-eb2fb9446ec2)
- start with "Issuing a full refund will close the inquiry...") to quickly inform users

--------

While looking at this, I realized the overall flow could use some tweaks. I don’t want to block progress on this, so I’m **happy to log these suggestions as a separate issue**, but here’s what stood out:

-     Having three buttons on a modal (2 CTAs + Cancel) isn’t great for usability.
-     The icons feel a bit unclear.
-     The wording and actions are a little vague, which might confuse users.

My suggestion is to reshape the modal slightly:

-     Let the user select a refund option directly in the modal.
-     Use just one CTA: Continue to Refund, which works for both options.

![Image](https://github.com/user-attachments/assets/ef6a421b-65ad-4410-a1a3-53f98a3f3540)


What do you think?

Also, in the spirit of aligning with WDS, we could update the modal to match their style, as I used in the screenshot above, ([Storybook link](https://wordpress.github.io/gutenberg/?path=/docs/components-modal--docs)) and use their radio buttons ([Storybook link](https://wordpress.github.io/gutenberg/?path=/docs/components-radiocontrol--docs)) too. Feels like a good chance to hit two birds with one stone!

Side note: I did think about combining the two steps for a full refund into one, but I feel it’s better to keep it as a focused two-step process.

![Image](https://github.com/user-attachments/assets/968809d0-909f-4aeb-b9a7-12f3b051616e)

**Comment by jessy-p** [comment]
Thanks @rogermattic  There is no hurry to move this to production, in this issue we will implement this modal as you suggested, with the simplified flow.

**Comment by rogermattic** [comment]
Hey @jessy-p ! Great! 


I reviewed the design above and decided to go for a **single-step option** so I made the final version of the design.

The [source is here in Figma](https://www.figma.com/design/lKQtvcEcbnjhz6EYbQaLW0/Refunds?node-id=1058-17811&t=duV4SCpyEtXlS7ZQ-1) 

And the flow looks as follows:

![Image](https://github.com/user-attachments/assets/1edc13eb-1998-4d6e-b741-0013b2d30c99)


Is this actionable for you and ready for the dev, or do you need anything else from me :) ?

**Comment by haszari** [comment]
Thanks for your work to improve the flow for partial disputes @rogermattic ! 

We had a chat about this in team hang and realised it's important to ship the initial, simpler fix; or an even simpler one. This basic flow focuses on the common case of a full refund, and allows merchants to use the order screen to partial refund if needed.

Here's what I'm proposing:

1. Ship a lean (basic) fix so the flow for the `Issue refund` button (the most prominent button/affordance) is no worse than the `Issue refund` menu item (buried in 3-dots menu). (See below for more details on this).
2. If we get feedback from merchants, or other evidence that we should make partial refunds easier (e.g. that they are more common than full), we can look at shipping an improved flow based on the new design.

#### Details for (1) – ship basic flow so `Issue refund` button is consistent

The idea here is that we focus on a quick flow for full refunds, reduce the number of steps for the common case.

- `Issue refund` button jumps straight to existing `Refund transaction` modal (which only does full refunds.
- Merchants who need to partially refund will need to either use three-dots menu or manually go to order screen. I'm betting this is less common.

<img width="435" alt="Image" src="https://github.com/user-attachments/assets/d51ed63c-2c61-4086-b1a9-c83a2d60fd7f" />

@jessy-p can you make that happen – i.e. rejig this PR as needed to ship the lean fix?

#### Thoughts on (2) (for future)

- Single-step / dialog flow is great. 
- Love that it leans on order screen for partial refund - defer to main UI.
- I'm assuming we'd use this flow for any entry point on the transaction screen e.g. 
  - `Issue a refund` button launches modal
  - Three dots menu has `Refund transaction…` and launches modal

This could even be considered for more-in-core; i.e. this is a good flow for refunding in WooCommerce, it shouldn't be limited to WooPayments transaction detail screen.

**Comment by Jinksi** [comment]
> @jessy-p can you make that happen – i.e. rejig this PR as needed to ship the lean fix?

@jessy-p I was about to pick this up but noticed the comment above about an existing PR. Do you have a WIP PR for this issue, or is this issue available for a new PR?

**Comment by jessy-p** [comment]
No, I made some local changes to get screenshots, no PR available.

**Comment by Jinksi** [comment]
No problem, thanks @jessy-p 🙏

**Comment by rogermattic** [comment]
OK thanks for following up on this @haszari !

>Here's what I'm proposing:

> Ship a lean (basic) fix so the flow for the Issue refund button (the most prominent button/affordance) is no worse than the Issue refund menu item (buried in 3-dots menu). (See below for more details on this).

Sounds good!

> Merchants who need to partially refund will need to either use three-dots menu or manually go to order screen. I'm betting this is less common.

I agree this is good for now, but will leave my concerns here for the future reference :)  
1. Not sure how discoverable the action under the three dots menu are
2. Having two entry points / placings for similar actions is not ideal :) 

Also, and more important – will we be able (after some time) review the number of clicks on each entry point? The number of actual refunds done (split by an entry point)? Do we collect that data?

> Thoughts on (2) (for future)
> Single-step / dialog flow is great.
> Love that it leans on order screen for partial refund - defer to main UI.
> I'm assuming we'd use this flow for any entry point on the transaction screen e.g.
> Issue a refund button launches modal
> Three dots menu has Refund transaction… and launches modal
> This could even be considered for more-in-core; i.e. this is a good flow for refunding in WooCommerce, it shouldn't be limited to WooPayments transaction detail screen.

Another thing is that the design features a refreshed modal style, which would be good to refresh in our UI (if we have any spare time :) ), too.
